infra/auth-server/src/utils/verify-sign.js

```diff
@@ -0,0 +1,44 @@
+'use strict'
+
+const Web3Utils = require('web3-utils')
+
+const Eth = require('web3-eth')
+const Web3Eth = new Eth()
+
+const stringify = require('json-stable-stringify')
+
+const logger = require('../logger')
+
+/**
+ * Verifies the sign of message
+ *
+ * @param {String} address ETH address of the signer
+ * @param {String} signature The actual signature
+ * @param {Object} payload The data that was signed
+ * @returns {Boolean} true if `signature` is signed by `address` on `payload` in the last 30 days; false otherwise
+ */
+const verifySign = ({ address, signature, payload }) => {
+  try {
+    // TODO: Should we also check if timestamp > Date.now()?
+    const minTimestamp = Date.now() - 1000 * 60 * 60 * 24 * 30
+    if (payload.timestamp < minTimestamp) {
```

Comment:
Including the timestamp in the signature(assuming it's verified) helps prevent a signature from being re-used.

Reply:
We should drastically reduce the period this eth signing timestamp is good for, perhaps to 20 minutes? Since there is no nonce, this allows generating an unlimited number of auth tokens during this period.

Reply:
> Perhaps somewhere we should have a comment about why we have one timestamp in the payload, and another roughly identical timestamp, also for 30 days, in the, in the JWT token itself?

Will do so.

> We should drastically reduce the period this eth signing timestamp is good for, perhaps to 20 minutes?

We could do that on Origin Marketplace app. But on other wallets and DApp browsers, I don't think we want the users to have the "Sign" prompt every hour or so.

But I agree, 30 days is a long time. We should perhaps reduce it.

Reply:
I'm just talking about how long the timestamp for generating new tokens is good for - not about the length of time that the token itself is good for.